### PR TITLE
Feature/tdp 2715 playground filter bar

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2715

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)

### CODE REVIEW

File: dataprep-webapp/src/app/components/filter/bar/filter-bar.html [hunk 1]
```diff
@@ -35,5 +30,12 @@
                      on-filter-change="filterBarCtrl.filterService.updateFilter(filter, value)"
                      on-filter-remove="filterBarCtrl.filterService.removeFilter(filter)"
                      enabled="filterBarCtrl.state.playground.filter.enabled"></filter-list>
+
+        <a id="reset-filters"
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/filter/bar/filter-bar.html]: Should be a button

File: dataprep-webapp/src/app/components/filter/monitor/_filter-monitor.scss [hunk 2]
```diff
@@ -12,26 +12,29 @@
...
+      padding-right: 10px;
+    }
   }
 
-  #reset-filters{
-    font-size: 0.7em;
-    color: $white;
+  #filter-monitor-right {
+    #filters-monitor-stats {
+      font-size: 0.9em;
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/filter/monitor/_filter-monitor.scss]: Please, stop using `em`

File: dataprep-webapp/src/app/components/filter/monitor/filter-monitor.html [hunk 3]
```diff
@@ -13,17 +13,23 @@
...
+    <div id="filter-monitor-left"
+         class="pull-left"
+         ng-if="filterMonitorCtrl.filters.length">
+        <span class="filters" translate-once="FILTERS"></span>
+        <label class="label-switch"
+               title="{{::'TOOGLE_ALL_FILTERS' | translate}}">
+            <input type="checkbox"
+                   ng-model="filterMonitorCtrl.state"
+                   ng-change="filterMonitorCtrl.onToogle()"/>
+            <div id="filter-toggle-switch" class="checkbox"></div>
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/filter/monitor/filter-monitor.html]: What's the role of this div?

File: dataprep-webapp/src/app/components/widgets/badge/_badge.scss [hunk 4]
```diff
@@ -11,22 +11,19 @@
 
   ============================================================================*/
 
-$badge-font-size: 0.85em !default;
+$badge-font-size: 1em !default;
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets/badge/_badge.scss]: No `em` please

File: dataprep-webapp/src/app/components/filter/monitor/filter-monitor.html (hunk 3 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/filter/monitor/filter-monitor.html]: It is used for the switch style recommanded by Maxime. See _switch.scss

Comment (frassinier) [inline dataprep-webapp/src/app/components/filter/monitor/filter-monitor.html]: Ok I don't really understand the role but it's not your change.

File: dataprep-webapp/src/app/components/filter/search/_filter-search.scss [hunk 5]
```diff
@@ -13,12 +13,19 @@
 
 #filter-search {
   input[type="search"] {
-    background-color: $white;
+    background-color: $shadow-gray;
     color: $dark-gray;
-    font-family: $base-font-family;
     font-size: 0.85em;
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/filter/search/_filter-search.scss]: Be careful with em unit

File: dataprep-webapp/src/app/components/playground/_playground.scss [hunk 6]
```diff
@@ -26,24 +26,38 @@ $slide-action-background: $base-background-color;
...
+      @include flex-shrink(1);
+      @include flex-basis(500px);
+
+      margin: 0 30px;
       position: relative;
+
+      .datagrid-container {
+        position: absolute;
+        width: 100%;
+        height: 100%;
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: With `position: absolute`, you should prefer `top: 0; left: 0; bottom: 0; right: 0;` to fit all the height of the relative parent

File: dataprep-webapp/src/app/components/filter/monitor/filter-monitor.html [hunk 7]
```diff
@@ -13,17 +13,23 @@
...
+<div id="filter-monitor">
+    <div id="filter-monitor-left"
+         class="pull-left"
+         ng-if="filterMonitorCtrl.filters.length">
+        <span class="filters" translate-once="FILTERS"></span>
+        <label class="label-switch"
+               title="{{::'TOOGLE_ALL_FILTERS' | translate}}">
+            <input type="checkbox"
+                   ng-model="filterMonitorCtrl.state"
+                   ng-change="filterMonitorCtrl.onToogle()"/>
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/filter/monitor/filter-monitor.html]: onToogle : enable/disable the filters when checking/unchecking the input checkbox

**Pull request merged**